### PR TITLE
Dynamic OLS

### PR DESCRIPTION
**Pull request by AustinJAdams**
I created a working Dynamic OLS.

I am going to go back and use more of the built-in ARCH modules, as I did my own lag picker and used StatsModels ADF tests instead of ARCH ones.  I would also like to allow people to use the other ARCH built-ins for unit-root checking and not just ADF. 

Now that I have the idea down, it shouldn't take long.

### CODE REVIEW

**Comment by bashtage** [comment]
Are the lgtm alerts valid?  

Looks very good on first glance. I'll give it a full pass tomorrow.

**Comment by AustinJAdams** [comment]
They were valid for the first ran commit, but I think I fixed them. If you like it, I will write some tests for it.

File: arch/unitroot/cointegration.py [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+        if reverse:
+            y0, y1 = y0[::-1], y1[::-1]
+
+        if _is_using_pandas(y0, y1):
+            columns = list(y1.columns)
+
+        else:
+            # Need to check if NumPy, because I can only support those two
+            n_obs, k = y1.shape
+            columns = [f'Var_{x}' for x in range(k)]
```
Comment (bashtage) [inline arch/unitroot/cointegration.py]: We still have Python 3.5, so need to use old style formatting.

**Comment by bashtage** [comment]
There is a syntax error at

arch/unitroot/cointegration.py:218:33: E999 SyntaxError: invalid syntax

**Comment by AustinJAdams** [comment]
I think that is also because of the use of f strings. I changed them to use .format(x) instead.

File: arch/unitroot/cointegration.py (hunk 1 above)
Comment (AustinJAdams) [reply arch/unitroot/cointegration.py]: Changed to .format()

**Pull request closed without merging**